src/blr/cli/buildmolecules.py

```diff
@@ -270,12 +271,17 @@ def write_molecule_stats(self, output_prefix, molecule_dict):
 
 
 def add_arguments(parser):
-    parser.add_argument("bam",
-                        help="Sorted BAM file tagged with barcode in the same tag as specified in -b/--barcode-tag.")
-    parser.add_argument("output",
-                        help="Output BAM file with molecule tags found under the tag specified at -m/--molecule-tag "
-                             "and molecules number for each barcode under the specified -n/--number-tag.")
-
+    parser.add_argument("input",
+                        help="Sorted SAM/BAM file tagged with barcode in the same tag as specified in "
+                             "-b/--barcode-tag.")
+
+    parser.add_argument("-o", "--output", default="-",
+                        help="Output SAM/BAM file with molecule tags found under the tag specified at "
+                             "-m/--molecule-tag and molecules number for each barcode under the specified "
+                             "-n/--number-tag. Default: write to stdout.")
+    parser.add_argument("-O", "--output-format", choices=["SAM", "BAM"], default="SAM",
```

Comment:
To me it makes more sense to infer name from output file and if `-O` is specified that is overridden.